client/landing/gutenboarding/onboarding-block/login-form/index.tsx

```diff
@@ -0,0 +1,123 @@
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
+import { Button, ExternalLink, TextControl, Modal, Notice } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { __experimentalCreateInterpolateElement } from '@wordpress/element';
+import { useI18n } from '@automattic/react-i18n';
+
+/**
+ * Internal dependencies
+ */
+import { AUTH_STORE } from '../../stores/auth';
+import './style.scss';
+
+// TODO: deploy this change to @types/wordpress__element
+declare module '@wordpress/element' {
+	// eslint-disable-next-line no-shadow
+	export function __experimentalCreateInterpolateElement(
+		interpolatedString: string,
+		conversionMap: Record< string, ReactElement >
+	): ReactNode;
+}
+
+const LoginForm = () => {
+	const { __: NO__, _x: NO_x } = useI18n();
+	const [ usernameOrEmailVal, setUsernameOrEmailVal ] = useState( '' );
+	const [ passwordVal, setPasswordVal ] = useState( '' );
+	const { submitUsernameOrEmail } = useDispatch( AUTH_STORE );
+	const { submitPassword } = useDispatch( AUTH_STORE );
+	const loginFlowState = useSelect( select => select( AUTH_STORE ).getLoginFlowState() );
+	const errors = useSelect( select => select( AUTH_STORE ).getErrors() );
+
+	// todo: loading state like isFetchingNewUser
+
+	const handleLogin = ( event: React.FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+
+		// todo: record analytics.
+		if ( loginFlowState === 'ENTER_USERNAME_OR_EMAIL' ) {
+			submitUsernameOrEmail( usernameOrEmailVal );
+		} else if ( loginFlowState === 'ENTER_PASSWORD' ) {
+			submitPassword( passwordVal );
+			//todo: handle success
+		}
```

Comment:
I think in the medium term we'll end up having a component for each `loginFlowState` and a parent component that renders one of them based on the current state. This does the trick though.